x/mongo/driver/connstring/connstring.go

```diff
@@ -376,6 +378,23 @@ func (p *parser) validateAuth() error {
 		if p.AuthMechanismProperties != nil {
 			return fmt.Errorf("MONGO-X509 cannot have mechanism properties")
 		}
+	case "mongodb-aws":
+		if p.Username != "" && p.Password == "" {
```

Comment:
Did you mean to perform these checks twice? Here and in the method `validateAndMakeCredentials`?

Reply:
@iwysiu This only outstanding question that I have. Otherwise, everything looks good to me.

Reply:
@bazile-clyde Sorry, I missed this earlier. It's duplicated because both the connstring and auth packages need to check credentials at separate times, and the way that the package dependencies work out connstring can't use auth package functions without a cyclical dependency.